src/Common/src/System/Net/LazyAsyncResult.cs

```diff
@@ -63,31 +63,46 @@ internal LazyAsyncResult(object myObject, object myState, AsyncCallback myCallBa
             _asyncState = myState;
             _asyncCallback = myCallBack;
             _result = DBNull.Value;
-            GlobalLog.Print("LazyAsyncResult#" + LoggingHash.HashString(this) + "::.ctor()");
+            if (GlobalLog.IsEnabled)
+            {
+                GlobalLog.Print("LazyAsyncResult#" + LoggingHash.HashString(this) + "::.ctor()");
+            }
         }
 
         // Allows creating a pre-completed result with less interlockeds.  Beware!  Constructor calls the callback.
         // If a derived class ever uses this and overloads Cleanup, this may need to change.
         internal LazyAsyncResult(object myObject, object myState, AsyncCallback myCallBack, object result)
         {
-            GlobalLog.Assert(result != DBNull.Value, "LazyAsyncResult#{0}::.ctor()|Result can't be set to DBNull - it's a special internal value.", LoggingHash.HashString(this));
+            if (result == DBNull.Value)
```

Comment:
I explicitly didn't because this seemed like an error condition that should really never happen and so it didn't seem worthwhile.  But I guess I can for consistency.